Sources/AWSSDKSwiftCore/MetaDataService.swift

```diff
@@ -111,16 +68,87 @@ struct MetaDataService {
         return futureResponse
     }
 
-    struct MetaData: Codable {
+    /// decode response return by metadata service
+    func decodeCredential(_ data: Data) -> CredentialProvider {
+        do {
+            let decoder = JSONDecoder()
+            // set JSON decoding strategy for dates
+            let dateFormatter = DateFormatter()
```

Comment:
no need for `ISO8601DateFormatter` since we don't use this service on non linux platforms anymore 👍